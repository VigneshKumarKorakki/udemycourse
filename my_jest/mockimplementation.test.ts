test("Mock function", () => {
  const myMockFun = jest.fn(cb => cb(null, true));
  
  myMockFun((err:null, val:boolean) => console.log(val)); 
  // OP:
  // true
})


// mockImplementation method is useful when you need to define the default implementation of a mock function that is created from another module

jest.mock('./foo')
const foo = require('./foo')

foo.mockImplementation(() => 42);
console.log(foo())
// OP:
42

// Mock names
test("Snapshot", () => {
  const myMockFn = jest
  .fn()
  .mockReturnValue('default')
  .mockImplementation(scalar => scalar + 42)
  .mockName('add42')

console.log(myMockFn(1))

expect(myMockFn).toMatchSnapshot()
}
)