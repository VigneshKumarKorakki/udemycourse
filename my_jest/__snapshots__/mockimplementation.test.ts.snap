// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
[MockFunction add42] {
  "calls": Array [
    Array [
      1,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": 43,
    },
  ],
}
`;
